src/dex/synthetix/synthetix-state.ts

```diff
@@ -0,0 +1,933 @@
+      const [tickCumulative, slot0] = result as [Record<0 | 1, bigint>, Slot0];
+
+      uniswapV3Slot0[address] = slot0;
+      tickCumulatives[address] = tickCumulative;
+    });
+
+    const [_packCounter, observationsRoundAndOverriddenCallData] =
+      this._buildObservationsRoundAndOverriddenCallData(
+        addressesFromPK,
+        uniswapV3Slot0,
+        aggregatorAddressesWithoutZeros,
+        latestRoundDatas,
+      );
+
+    const [block, observationsRoundDataAndOverridden] = await Promise.all([
+      this.dexHelper.web3Provider.eth.getBlock(blockNumber || 'latest'),
+      this.multiWrapper.tryAggregate<
+        OracleObservation | LatestRoundData | string
+      >(true, observationsRoundAndOverriddenCallData, blockNumber),
+    ]);
+
+    addressesFromPKBoundary = addressesFromPK.length * packCounter;
+    const observations = observationsRoundDataAndOverridden
+      .slice(0, addressesFromPKBoundary)
+      .map(e => e.returnData) as OracleObservation[];
+
+    const overriddenPools = observationsRoundDataAndOverridden
+      .slice(
+        addressesFromPKBoundary,
+        addressesFromPKBoundary + this.onchainConfigValues.poolKeys.length,
+      )
+      .map(e => e.returnData) as string[];
+
+    const otherRoundDatas = observationsRoundDataAndOverridden
+      .slice(addressesFromPKBoundary + this.onchainConfigValues.poolKeys.length)
+      .map(e => e.returnData) as LatestRoundData[];
+
+    const chunkedOtherRoundDatas =
+      this.onchainConfigValues.exchangeDynamicFeeConfig.rounds > 0n
+        ? _.chunk(
+            otherRoundDatas,
+            Number(this.onchainConfigValues.exchangeDynamicFeeConfig.rounds),
+          )
+        : new Array<LatestRoundData[]>(latestRoundDatas.length).fill([]);
+
+    const overriddenPoolForRoute = overriddenPools.reduce<
+      Record<string, Address>
+    >((acc, curr, i) => {
+      acc[
+        dexPriceAggregatorUniswapV3.identifyRouteFromPoolKey(
+          this.onchainConfigValues!.poolKeys[i],
+        )
+      ] = curr;
+      return acc;
+    }, {});
+
+    _require(
+      latestRoundDatas.length === chunkedOtherRoundDatas.length,
+      'Something is wrong with indexes while getting OnchainState',
+      { chunkedOtherRoundDatas, latestRoundDatas },
+      'latestRoundDatas.length === chunkedOtherRoundDatas.length',
+    );
+
+    const aggregators = latestRoundDatas.reduce<Record<Address, ChainlinkData>>(
+      (acc, cur, i) => {
+        const getRoundData = chunkedOtherRoundDatas[i].reduce<
+          Record<BigIntAsString, LatestRoundData>
+        >((roundAcc, currRound) => {
+          roundAcc[currRound.roundId.toString()] = currRound;
+          return roundAcc;
+        }, {});
+
+        acc[Object.keys(aggregatorAddressesWithoutZeros)[i]] = {
+          latestRoundData: cur,
+          getRoundData,
+        };
+        return acc;
+      },
+      {},
+    );
+
+    const uniswapV3Observations = _.chunk(observations, packCounter).reduce<
```

Comment:
Again, should be `_packCounter` not `packCounter` I believe.

Reply:
Good catch, thanks! Fixed it 👍